app/database_etl/test_adjustment_handler/test_adjustment.py

```diff
@@ -0,0 +1,246 @@
+
+        # Validate model_params using marshmallow
+        try:
+            ModelParamsSchema().load(model_params)
+        except ValidationError as err:
+            print("Error: ", err.messages)
+            return None, None, None
+
+        satisfactory_model_set_found = False
+        n_modelsets = 0
+
+        while not satisfactory_model_set_found:
+            satisfactory_model_set = []
+            for _ in range(self.n_replicates):
+                n_trials = 0
+                satisfactory_model_found = False
+
+                while not satisfactory_model_found:
+                    # Attempt to fit a model
+                    summary_df_parsed, satisfactory_model_found = self.fit_one_pystan_model(model_params)
+                    # If number of attempts exceeded, return Nones
+                    n_trials += 1
+                    if n_trials >= self.trials_lim:
+                        print('no models met the HMC diagnostics in {trials_lim} trials')
+                        return None, None, None
+
+                satisfactory_model_set.append(summary_df_parsed)
+
+            satisfactory_models_df = pd.DataFrame(satisfactory_model_set)
+
+            # see whether all results are within 10% of median; if not, rerun
+            model_medians = satisfactory_models_df['50%']
+            median = model_medians.median()
+            consistent_results_found = model_medians.between(0.9 * median, 1.1 * median).all()
+            median_run = satisfactory_models_df[satisfactory_models_df['50%'] == median].iloc[0]
+
+            lower, upper = arviz.hdi(median_run['samples'], credible_interval_size)
+            best_fit = median_run['fit']
+
+            try:
+                raw_prev = model_params['y_prev_obs'] / model_params['n_prev_obs']
+                log_delta = log(median + 1e-3, 10) - log(raw_prev + 1e-3, 10)
+                median_is_bounded = abs(log_delta) < 1
+            except (ZeroDivisionError, ValueError):
+                return None, None, None
+
+            satisfactory_model_set_found = consistent_results_found and median_is_bounded
+
+            n_modelsets += 1
+            if n_modelsets > self.modelsets_lim:
+                return None, None, None
+
+        if self.return_fit:
+            return best_fit
+        else:
+            return lower, median, upper
+
+    def get_adjusted_estimate(self, estimate: pd.Series) -> Tuple:
+        # initialize adj_type
+        adj_type = None
+
+        # unadjusted estimate available and thus prioritized in estimate selection code
+        if pd.isna(estimate['test_adj']):
+
+            # Independent evaluation is available
+            if pd.notna(estimate['ind_se']) and pd.notna(estimate['ind_sp']):
+                adj_type = 'FINDDx / MUHC independent evaluation'
+                # Also note these must be divided by 100
+                se = (estimate['ind_se']) / 100
```

Comment:
Don't think we need to multiply and divide by 100 anymore now that the ETL cleanup tickets that rahul requested have been merged. But can you verify @simonarocco